tests/functional_test_base.go

```diff
@@ -415,3 +418,8 @@ func (s *FunctionalTestBase) registerArchivalNamespace(archivalNamespace string)
 	)
 	return err
 }
+
+// MatchingServiceClient returns the matching service client setup for this functional test.
+func (s *FunctionalTestBase) MatchingServiceClient() matchingservice.MatchingServiceClient {
```

Comment:
Can we do the history client similarly? Both are in temporalImpl and TestCluster so if you're adding another layer, let's do it to both?